zipline/algorithm.py

```diff
@@ -986,6 +995,15 @@ def set_commission(self, commission):
             raise OverrideCommissionPostInit()
         self.commission = commission
 
+    @api_method
+    def set_symbol_lookup_date(self, dt):
+        """
+        Set the date for which symbols will be resolved to their sids
+        (symbols may map to different firms or underlying assets at
+        different times)
+        """
+        self.symbol_lookup_date = dt
```

Comment:
You could just say `pd.Timestamp(dt)`, which will accept datetimes and Timestamps, plus strings, and take care of the check for you.